src/app.py

```diff
@@ -1,6 +1,11 @@
 import os
 import pyaudio
 from piper import PiperVoice
+from gpiozero import Button
+from time import sleep
+
+BUTTON_PIN = 17
+button = Button(BUTTON_PIN)
```

Comment:
```
/home/ats/projects/venvs/kitchen_assistant/lib/python3.9/site-packages/gpiozero/devices.py:300: PinFactoryFallback: Falling back from lgpio: No module named 'lgpio'
  warnings.warn(
/home/ats/projects/venvs/kitchen_assistant/lib/python3.9/site-packages/gpiozero/devices.py:300: PinFactoryFallback: Falling back from rpigpio: No module named 'RPi'
  warnings.warn(
/home/ats/projects/venvs/kitchen_assistant/lib/python3.9/site-packages/gpiozero/devices.py:300: PinFactoryFallback: Falling back from pigpio: No module named 'pigpio'
  warnings.warn(
/home/ats/projects/venvs/kitchen_assistant/lib/python3.9/site-packages/gpiozero/devices.py:297: NativePinFactoryFallback: Falling back to the experimental pin factory NativeFactory because no other pin factory could be loaded. For best results, install RPi.GPIO or pigpio. See https://gpiozero.readthedocs.io/en/stable/api_pins.html for more information.
```

Reply:
```
As above, on a Raspberry Pi with the RPi.GPIO library installed, (assuming no environment variables are set), the default pin factory will be [RPiGPIOFactory](https://gpiozero.readthedocs.io/en/stable/api_pins.html#gpiozero.pins.rpigpio.RPiGPIOFactory).
```

> https://gpiozero.readthedocs.io/en/stable/api_pins.html

Reply:
- Raspberry Pi has the RPi.GPIO library as default
```
(kitchen_assistant) ats@raspberrypi:~/projects/kitchen_assistant $ pip list
Package             Version
------------------- --------------
coloredlogs         15.0.1
colorzero           2.0
flatbuffers         20181003210633
gpiozero            2.0.1
humanfriendly       10.0
importlib-metadata  4.13.0
importlib-resources 5.13.0
mpmath              1.3.0
numpy               1.26.4
onnxruntime         1.18.0
packaging           24.0
pip                 20.3.4
piper-phonemize     1.1.0
piper-tts           1.2.0
pkg-resources       0.0.0
protobuf            5.27.0
PyAudio             0.2.14
setuptools          44.1.1
sympy               1.12.1
zipp                3.19.0
(kitchen_assistant) ats@raspberrypi:~/projects/kitchen_assistant $ deactivate
ats@raspberrypi:~/projects/kitchen_assistant $ pip list
Package           Version
----------------- ---------
arandr            0.1.10
astroid           2.5.1
asttokens         2.0.4
automationhat     0.2.0
beautifulsoup4    4.9.3
blinker           1.4
blinkt            0.1.2
buttonshim        0.0.2
Cap1xxx           0.1.3
certifi           2020.6.20
chardet           4.0.0
click             7.1.2
colorama          0.4.4
colorzero         1.1
cryptography      3.3.2
cupshelpers       1.0
dbus-python       1.2.16
distro            1.5.0
docutils          0.16
drumhat           0.1.0
envirophat        1.0.0
ExplorerHAT       0.4.2
Flask             1.1.2
fourletterphat    0.1.0
gpiozero          1.6.2
html5lib          1.1
idna              2.10
isort             5.6.4
itsdangerous      1.1.0
jedi              0.18.0
Jinja2            2.11.3
lazy-object-proxy 0.0.0
logilab-common    1.8.1
lxml              4.6.3
MarkupSafe        1.1.1
mccabe            0.6.1
microdotphat      0.2.1
mote              0.0.4
motephat          0.0.3
mypy              0.812
mypy-extensions   0.4.3
numpy             1.19.5
oauthlib          3.1.0
pantilthat        0.0.7
parso             0.8.1
pexpect           4.8.0
pgzero            1.2
phatbeat          0.1.1
pianohat          0.1.0
picamera2         0.3.12
pidng             4.0.9
piexif            1.1.3
piglow            1.2.5
pigpio            1.78
Pillow            8.1.2
pip               20.3.4
psutil            5.8.0
pycairo           1.16.2
pycups            2.0.1
pygame            1.9.6
Pygments          2.7.1
PyGObject         3.38.0
pyinotify         0.9.6
PyJWT             1.7.1
pylint            2.7.2
PyOpenGL          3.1.5
pyOpenSSL         20.0.1
PyQt5             5.15.2
PyQt5-sip         12.8.1
pyserial          3.5b0
pysmbc            1.0.23
python-apt        2.2.1
python-prctl      1.7
rainbowhat        0.1.0
reportlab         3.5.59
requests          2.25.1
requests-oauthlib 1.0.0
responses         0.12.1
roman             2.0.0
RPi.GPIO          0.7.0
RTIMULib          7.2.1
scrollphat        0.0.7
scrollphathd      1.2.1
Send2Trash        1.6.0b1
sense-hat         2.6.0
setuptools        52.0.0
simplejpeg        1.6.4
simplejson        3.17.2
six               1.16.0
skywriter         0.0.7
sn3218            1.2.7
soupsieve         2.2.1
spidev            3.5
ssh-import-id     5.10
thonny            4.0.1
toml              0.10.1
touchphat         0.0.1
twython           3.8.2
typed-ast         1.4.2
typing-extensions 3.7.4.3
unicornhathd      0.0.4
urllib3           1.26.5
v4l2-python3      0.3.2
webencodings      0.5.1
Werkzeug          1.0.1
wheel             0.34.2
wrapt             1.12.1
```

Reply:
Test code
> https://forums.balena.io/t/badpinfactory-error-when-using-gpiozero-python-library-with-raspberrypi-4/367262/3